cmd/cosign/cli/sign.go

```diff
@@ -153,21 +157,25 @@ func SignCmd(ctx context.Context, so SignOpts,
 	if err != nil {
 		return errors.Wrap(err, "getting remote image")
 	}
+
 	repo := ref.Context()
 	img := repo.Digest(get.Digest.String())
-	// The payload can be specified via a flag to skip generation.
-	var payload []byte
-	if payloadPath != "" {
-		fmt.Fprintln(os.Stderr, "Using payload from:", payloadPath)
-		payload, err = ioutil.ReadFile(filepath.Clean(payloadPath))
-	} else {
-		payload, err = (&sigPayload.Cosign{
-			Image:       img,
-			Annotations: so.Annotations,
-		}).MarshalJSON()
-	}
-	if err != nil {
-		return errors.Wrap(err, "payload")
+
+	toSign := []name.Digest{img}
+
+	if recursive && get.MediaType.IsIndex() {
+		idx, err := get.ImageIndex()
+		if err != nil {
+			return err
+		}
+		idxManifest, err := idx.IndexManifest()
+		if err != nil {
+			return err
+		}
+		for _, manifest := range idxManifest.Manifests {
```

Comment:
I always forget - is it possible to create an image that recurses even further? An index of indexes?

Reply:
not iiuc @jonjohnsonjr

Reply:
it is

Reply:
Talked with him and apparently recursive indexes _can_ exist 🤮

Reply:
walking the tree now

Reply:
it hurts me that the flag is recursive but the implementation is not

Reply:
your anguish sustains me ❤️